amgcl/backend/vexcl.hpp

```diff
@@ -302,7 +302,12 @@ struct residual_impl<
     vex::sparse::distributed<vex::sparse::matrix<Va,C,P>>,
     vex::vector<Vf>,
     vex::vector<Vx>,
-    vex::vector<Vr>
+    vex::vector<Vr>,
```

Comment:
I would expect the specialization in `vexcl_static_matrix.hpp` to be more precise and thus to have precedence over this specialization. Do you get ambiguous template instantiation errors without the next enable_if?

Reply:
Also, does this work with `V*=static_matrix<T,1,1>`?

Reply:
Maybe, I don't know (I compute the residual error using an existing in-house implementation, so the compiler doesn't try to instantiate the template). But since I had a similar issue with `spmv_impl` after adding the two new template parameters (`C` and `P`) in `vexcl_static_matrix.hpp` I though that the same would occur with the residual implementation and thus I preferred to be on de safe side.

Reply:
ok, that makes sense